lib/virt_autotest/virtual_network_utils.pm

```diff
@@ -169,8 +168,20 @@ sub restore_guests {
         script_run("virsh destroy $_");
         script_run("virsh undefine $_");
         script_run("virsh define /tmp/$_.xml");
-        script_run("rm -rf /tmp/$_.xml");
-        script_run("sed -i '/$_/d' /etc/hosts");
+    }
+}
+
+sub reset_guests {
+    return if get_var('INCIDENT_ID');    # QAM does not recreate guests every time
+    my $get_vm_hostnames   = "virsh list  --all | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames       = script_output($get_vm_hostnames, 30, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_array = split(/\n+/, $vm_hostnames);
+    foreach (@vm_hostnames_array)
+    {
+        script_run("virsh destroy $_");
+        script_run("virsh undefine $_");
+        script_run("virsh define /tmp/$_.xml");
+        script_run("virsh start $_");
```

Comment:
Could you help implement this function as a more general one that can be used by others as well ?  For example, Invent an argument "xml file folder", then call reset_guests("/tmp"). Then move this to virt_utils.pm ?

Reply:
I agree with to have a common function as well. SR-IOV tests need fresh guest at the begining, then it modifies the vm xml and vm disk during test processing, the subsequent test then need fresh guests. But would it be better to put the initiation in base test module(virt_feature_test_base) instead of virt_utils?

Reply:
Great, got it. 
will update this changed soon.

Reply:
talked with Leon offline, it does not intend to initialize a guest. it is just to redefine a guest with a specified vm xml. So agree to put it in utils. But I still call for a guest initialization function in base test as each test need it.

Reply:
Or determine a fixed location to put vm xml, determine in which state of the guest are when test begins.

Reply:
@waynechen55 as your mentioned, created recreate_guests func under virt_utils. 
Please check with it from virt_utils.